bigchaindb/util.py

```diff
@@ -436,19 +439,22 @@ def validate_fulfillments(signed_transaction):
     Returns:
         bool: True if the signature is correct, False otherwise.
     """
-
     for fulfillment in signed_transaction['transaction']['fulfillments']:
         fulfillment_message = get_fulfillment_message(signed_transaction, fulfillment)
         try:
             parsed_fulfillment = cc.Fulfillment.from_uri(fulfillment['fulfillment'])
         except (TypeError, ValueError, ParsingError):
             return False
+
         is_valid = parsed_fulfillment.validate(serialize(fulfillment_message))
 
         # if transaction has an input (i.e. not a `CREATE` transaction)
-        if fulfillment['input']:
-            is_valid &= parsed_fulfillment.condition.serialize_uri() == \
-                fulfillment_message['condition']['condition']['uri']
+        # TODO: avoid instantiation, pass as argument!
+        bigchain = bigchaindb.Bigchain()
+        input_condition = get_input_condition(bigchain, fulfillment)
+        is_valid &= parsed_fulfillment.condition_uri == \
```

Comment:
I'd use the logical operator `and` instead of the bitwise operator `&`, so:
```python
is_valid = is_valid and parsed_fulfillment.condition_uri == input_condition['condition']['uri']
```
because we are doing a logical operation, and not working on a bit level.
AFAIK your solution with `&=` and my proposal with `and` are totally compatible, it's just to be a bit (pun intended) more precise :santa: